### PR TITLE
Set thread names for the Win32 platform

### PR DESCRIPTION
**Pull request by mrpippy**
Set thread names for the Win32 platform.

I've tested this in Wine, but not on actual Windows.

### CODE REVIEW

**Review by flibitijibibo — APPROVED** [review]
Looks right to me! I don't think we need to try and implement the hacky pre-Win10 thread naming scheme so this should be fine.

Will merge when this gets a +1 from another platform_win32 maintainer, just in case.

**Comment by rbernon** [comment]
I think loading `SetThreadDescription` could go in `FAudio_PlatformInit` instead of requiring a specific InitOnce, and if we take a kernelbase module ref we should probably release it in `FAudio_PlatformQuit`. Otherwise looks good.

**Comment by mrpippy** [comment]
Thanks, good point, I didn't realize only one mastering voice was allowed at a time. I pushed a new version with that change and some other parts tightened up.

**Review by rbernon — APPROVED** [review]
LGTM

**Pull request merged**
